src/lib/yang/alarms.lua

```diff
@@ -1,5 +1,53 @@
 module(..., package.seeall)
 
+local data = require('lib.yang.data')
+local util = require('lib.yang.util')
+
+local state = {
+   alarm_inventory = {},
+}
+
+function get_state ()
+   return {
+      alarm_inventory = state.alarm_inventory,
+   }
+end
+
+-- Single point to access alarm type keys.
+alarm_type_keys = {}
+
+function alarm_type_keys:fetch (...)
+   self.cache = self.cache or {}
+   local function lookup (alarm_type_id, alarm_type_qualifier)
+      if not self.cache[alarm_type_id] then
+         self.cache[alarm_type_id] = {}
+      end
+      return self.cache[alarm_type_id][alarm_type_qualifier]
+   end
+   local alarm_type_id, alarm_type_qualifier = unpack({...})
```

Comment:
Why not just specify alarm_type_id and alarm_type_qualifier as formal arguments?